File: Golang/AddTwoNumbers/AddTwoNumbers_test.go
package AddTwoNumbers

import (
	"testing"
	"reflect"
)

func TestAddTwoNumbers(t *testing.T) {

	if reflect.DeepEqual(addTwoNumbers(
		&ListNode{
			Val: 0,
		}, &ListNode{
			Val: 0,
		}),
		&ListNode{
			Val: 0,
		}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(addTwoNumbers(
		&ListNode{
			Val: 5,
		}, &ListNode{
			Val: 5,
		}),
		&ListNode{
			Val: 0,
			Next: &ListNode{
				Val: 1,
			},
		}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(addTwoNumbers(
		&ListNode{
			Val: 2,
			Next: &ListNode{
				Val: 4,
				Next: &ListNode{
					Val: 3,
				},
			},
		}, &ListNode{
			Val: 5,
			Next: &ListNode{
				Val: 6,
				Next: &ListNode{
					Val: 4,
				},
			},
		}),
		&ListNode{
			Val: 7,
			Next: &ListNode{
				Val: 0,
				Next: &ListNode{
					Val: 8,
				},
			},
		}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(addTwoNumbers(
		&ListNode{
			Val: 1,
		}, &ListNode{
			Val: 9,
			Next: &ListNode{
				Val: 9,
			},
		}),
		&ListNode{
			Val: 0,
			Next: &ListNode{
				Val: 0,
				Next: &ListNode{
					Val: 1,
				},
			},
		}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/AddTwoNumbersTest.py
from unittest import TestCase

from AddTwoNumbers import AddTwoNumbers, ListNode


class TestAddTwoNumbers(TestCase):
    def test_addTwoNumbers(self):
        atn = AddTwoNumbers()
        node0 = ListNode(0)
        node5 = ListNode(5)
        self.assertTrue(atn.addTwoNumbers(node0, node0) == ListNode(0))
        tmp = ListNode(0)
        tmp.next = ListNode(1)
        self.assertEqual(atn.addTwoNumbers(node5, node5), tmp)

        node243 = ListNode(2)
        node243.next = ListNode(4)
        node243.next.next = ListNode(3)

        node564 = ListNode(5)
        node564.next = ListNode(6)
        node564.next.next = ListNode(4)

        tmp = ListNode(7)
        tmp.next = ListNode(0)
        tmp.next.next = ListNode(8)
        self.assertEqual(atn.addTwoNumbers(node243, node564), tmp)

        node1 = ListNode(1)
        node99 = ListNode(9)
        node99.next = ListNode(9)
        tmp = ListNode(0)
        tmp.next = ListNode(0)
        tmp.next.next = ListNode(1)
        self.assertEqual(atn.addTwoNumbers(node1, node99), tmp)
